runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -253,15 +258,33 @@ local function on_line_impl(self, buf, line)
       if capture == nil then break end
 
       local start_row, start_col, end_row, end_col = node:range()
-      local hl = highlighter_query.hl_cache[capture]
-
-      if hl and end_row >= line then
-        a.nvim_buf_set_extmark(buf, ns, start_row, start_col,
-                               { end_line = end_row, end_col = end_col,
-                                 hl_group = hl,
-                                 ephemeral = true,
-                                 priority = tonumber(metadata.priority) or 100 -- Low but leaves room below
-                                })
+
+      if metadata.sub_captures then
+        for _, sub_capture in ipairs(metadata.sub_captures) do
+          local sub_capture_name, range = unpack(sub_capture)
+          start_row, start_col, end_row, end_col = unpack(range)
+          local capture_name = highlighter_query._query.captures[capture]
+          local hl = highlighter_query.hl_cache[capture_name..'.'..sub_capture_name]
+
+          if hl and end_row >= line then
+            a.nvim_buf_set_extmark(buf, ns, start_row, start_col,
+                                   { end_line = end_row, end_col = end_col,
+                                     hl_group = hl,
+                                     ephemeral = true,
+                                     priority = tonumber(metadata.priority) or 100 -- Low but leaves room below
+                                    })
+          end
+        end
+      else
```

Comment:
I'm having problems to understand this else here, so there will be no normal highlight and only the highlights of the subcaptures when there are subcaptures?

Reply:
Yeah I guess it makes sense to always do this branch. I'll change.